src/core/components/IconButton/IconButton.tsx

```diff
@@ -2,21 +2,25 @@ import IconButton from '@mui/material/IconButton';
 import { MouseEvent } from 'react';
 import './IconButton.css';
 
-interface IconButtonProps {
+interface CustomIconButtonProps {
   icon: JSX.Element;
-  label: string;
-  isSelected: boolean;
-  id: string;
+  label?: string;
+  variant?: 'primary' | 'secondary';
   handleClick: (event: MouseEvent<HTMLButtonElement>) => void;
 }
 
-export const CustomIconButton = ({ icon, label, isSelected, id, handleClick }: IconButtonProps) => {
+export const CustomIconButton = ({ icon, label, variant = 'primary', handleClick }: CustomIconButtonProps) => {
+  const validVariants = ['primary', 'secondary'];
+  const defaultVariant = 'primary';
+
+  variant = validVariants.includes(variant) ? variant : defaultVariant;
```

Comment:
Please define default styles for the default class ('icon-button') . Then we dont need to use all of these validations. The secondary styles should overwrite default one if a corresponding variant is passed to the props